### PR TITLE
Param passing prototype: Modify action

### PR DESCRIPTION
**Pull request by jennuine**
Keeping as draft until https://github.com/osrf/sdformat/pull/566 is merged

# 🎉 New feature

## Summary
Implementation of the `modify` action from the [param passing proposal](http://sdformat.org/tutorials?tut=param_passing_proposal&branch=param_passing). Gives users the ability to modify elements and/or attributes from the included model.

~~A limitation is that the resulting include file is not validated and it will be up to the user to have modified elements correctly specified. Once this is transitioned out of the `experimental` phase (and into a core feature), this will be addressed.~~
The resulting include file is validated in the [`addNestedModel` call](https://github.com/osrf/sdformat/blob/1398d15a15b6693901e1ede77353443d89af26c9/src/parser.cc#L1096) after the params have been updated (`addNestedModel` eventually calls `readXml` on the updated included model). This will no longer be the case in `sdf11` but the same logic can be applied (i.e., calling `readXml` on the included model after the params have been updated).

## Test it
`./build/sdformat10/test/integration/INTEGRATION_param_passing`

## Checklist
- [X] Signed all commits for DCO
- [X] Added tests
- [x] Added example and/or tutorial: osrf/sdf_tutorials#51
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [X] Code check passed (In source directory, run `sh tools/code_check.sh`)
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [X] While waiting for a review on your PR, please help review
[another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+)
to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

**Comment by jennuine** [comment]
For a tutorial see osrf/sdf_tutorials#51

File: src/ParamPassing.cc [hunk 1]
```diff
@@ -506,6 +526,103 @@ void add(tinyxml2::XMLElement *_childXml, ElementPtr _elem, Errors &_errors)
...
+//////////////////////////////////////////////////
+void modifyAttributes(tinyxml2::XMLElement *_xml,
+                      ElementPtr _elem, Errors &_errors)
+{
+  for (const tinyxml2::XMLAttribute *attr = _xml->FirstAttribute();
+      attr; attr = attr->Next())
+  {
+    std::string attrName = attr->Name();
+
+    if (attrName == "element_id" || attrName == "action") continue;
```
Comment (iche033) [inline src/ParamPassing.cc:538]: `continue` on new line

File: test/integration/include_custom_model.sdf [hunk 2]
```diff
@@ -138,6 +141,55 @@
...
+                <far>50</far>
+                <near>0.05</near>
+              </clip>
+              <horizontal_fov/>
+            </camera>
+          </sensor>
+
+          <!-- <sensor element_id="chassis::camera" action="modify">
+            <pose relative_to=""/>
+          </sensor> -->
```
Comment (iche033) [inline test/integration/include_custom_model.sdf:163]: remove these lines?

**Review by iche033 — APPROVED** [review]
looks good to me. Just a couple of minor comments.

File: src/ParamPassing.cc (hunk 1 above)
Comment (jennuine) [reply src/ParamPassing.cc]: 49b0b04b90bb5d5df2ddbbf4246b5cbe4076e847

File: test/integration/include_custom_model.sdf (hunk 2 above)
Comment (jennuine) [reply test/integration/include_custom_model.sdf]: Thanks 49b0b04b90bb5d5df2ddbbf4246b5cbe4076e847

**Pull request merged**